File: TradeHarborApi/Models/PostFeatures/PostComment.cs
namespace TradeHarborApi.Models.PostFeatures
{
    /// <summary>
    /// Represents a comment entity associated with a post.
    /// </summary>
    public class PostComment
    {
        /// <summary>
        /// Gets or sets the unique identifier of the comment.
        /// </summary>
        public string CommentId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the unique identifier of the post to which the comment belongs.
        /// </summary>
        public string PostId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the content of the comment.
        /// </summary>
        public string Comment { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the timestamp when the comment was created.
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Gets or sets the username of the user who created the comment.
        /// </summary>
        public string Username { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the URL of the user's profile picture.
        /// </summary>
        public string ProfilePictureUrl { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the first name of the user who created the comment.
        /// </summary>
        public string FirstName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the last name of the user who created the comment.
        /// </summary>
        public string LastName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the user ID of the user who created the comment.
        /// </summary>
        public string UserId { get; set; } = string.Empty;
    }
}

File: TradeHarborClient/src/app/models/api-responses/postComment.ts
/**
 * Represents a comment on a post with user information.
 */
export class PostComment {
    /**
     * The unique identifier for the comment.
     */
    commentId: string = '';
  
    /**
     * The unique identifier for the post to which the comment belongs.
     */
    postId: string = '';
  
    /**
     * The content of the comment.
     */
    comment: string = '';
  
    /**
     * The timestamp when the comment was created.
     */
    timestamp: string = '';
  
    /**
     * The username of the user who made the comment.
     */
    username: string = '';
  
    /**
     * The URL of the user's profile picture.
     */
    profilePictureUrl: string = '';
  
    /**
     * The first name of the user who made the comment.
     */
    firstName: string = '';
  
    /**
     * The last name of the user who made the comment.
     */
    lastName: string = '';
  
    /**
     * The unique identifier for the user who made the comment.
     */
    userId: string = '';
  }
  